### PR TITLE
update circleci config

### LINKED ISSUE #3978: Update circle ci python image to be consistent with runtime version
**What we're after:**
Current runtime environment is 3.7.8 as specified by 3.7.x in runtime.txt. However, current circle ci image is 3.7.4 per .config.yml.

Let's make these consistent so that the tests run are with the same version as the runtime environment for our app.

### Completion criteria
- [ ] circle ci version are updated to match what's in runtime 

### Related issues
- #522
- #3978 
- #https://github.com/fecgov/openFEC/issues/4425

### PR DESCRIPTION
**Pull request by pkfec**
## Summary (required)

1. Update python docker image version to match the python version given in the runtime.txt file
2. Update Postgres docker image version to match the DB version in our cloud.gov (medium-gp-psql-redundant) environment
3. Set POSTGRES_HOST_AUTH_METHOD: "trust" environment variable. Builds fail without this env variable. 
 See: 
https://discuss.circleci.com/t/postgresql-image-password-not-specified-issue/34555?utm_medium=SEM&utm_source=gnb&utm_campaign=SEM-gb-DSA-Eng-uscan&utm_content=&utm_term=dynamicSearch-&gclid=CjwKCAjw_NX7BRA1EiwA2dpg0hsoFeu7z02id7KLIrJiImZpUfDShDzw7DbAwJJdL75bstnqP4914RoCoWAQAvD_BwE

- Resolves #3978 
_Include a summary of proposed changes._

- Update docker images in **circleci/config.yml file** : 

1. Python  from 3.7.4-stretch-node-browsers to 3.7.9-stretch-node-browsers
2. Postgres from 9.6.8 to 11.9
3. Add script to accommodate node upgrades in circleci

## Screenshots
Before upgrade:
Screen Shot 2020-10-02 at 10 06 47 AM

Screen Shot 2020-10-02 at 10 07 06 AM

After upgrade:
Screen Shot 2020-10-06 at 12 57 02 PM

Screen Shot 2020-10-02 at 10 05 46 AM

## How to test the changes ~locally~
 - Rebuild or Rerun _feature/3978-update-circleci-config_ branch on circleci  and notice  **Spin up  environment** and **Container** circleci tasks downloading the upgraded python and postgres docker images.

### CODE REVIEW

**Review by johnnyporkchops — APPROVED** [review]
Looks good Priya! This will make life easier for local CMS development!

File: .circleci/config.yml [hunk 1]
```diff
@@ -50,8 +51,16 @@ jobs:
...
+            curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.6/install.sh | bash
+            echo ". ~/.nvm/nvm.sh" >> $BASH_ENV
+            export NVM_DIR="$HOME/.nvm"
+            [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+            [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+            nvm install 10.16.0
+            nvm use 10.16.0
+            nvm alias default 10.16.0
             sudo npm install -g npm
             npm install webpack
```
Comment (rfultz) [inline .circleci/config.yml:65]: Outside the scope of this ticket, but we don't need `npm install webpack` if we also have `npm install` (and it's on the next line)

File: .circleci/config.yml [hunk 2]
```diff
@@ -50,8 +51,16 @@ jobs:
             pip install -r requirements.txt
 
       - run:
-          name: Install Node.js dependencies
+          name: Install node dependencies
```
Comment (rfultz) [inline .circleci/config.yml:54]: Nowhere near a dealbreaker but "Install Node dependencies" doesn't really describe what's happening here. The node packages are being installed, yes, but it's also building the entire front-end of the site (JavaScript, CSS/scss, image copying)

File: .circleci/config.yml [hunk 3]
```diff
@@ -9,15 +9,16 @@ jobs:
...
       # Python
-      - image: circleci/python:3.7.4-stretch-node-browsers
+      - image: circleci/python:3.7.9-stretch-node-browsers
         environment:
           TZ: America/New_York
           DATABASE_URL: postgres://postgres@0.0.0.0/cfdm_cms_test
 
       # PostgreSQL
-      - image: circleci/postgres:9.6.8
+      - image: circleci/postgres:11.9
```
Comment (rfultz) [inline .circleci/config.yml:18]: In the build, why did this section end with `Build was canceled`?

File: .circleci/config.yml [hunk 4]
```diff
@@ -9,15 +9,16 @@ jobs:
...
         environment:
           POSTGRES_USER: postgres
+          POSTGRES_HOST_AUTH_METHOD: "trust"
```
Comment (rfultz) [inline .circleci/config.yml:21]: The build is giving us a warning about this line (POSTGRES_HOST_AUTH_METHOD). Is this something we want to do outside of local? (The warning looks important but I'd rely 100% on others' expertise)

File: .circleci/config.yml (hunk 3 above)
Comment (pkfec) [reply .circleci/config.yml:18]: @rfultz - Its a circleci bug that is around for a while. 
See here : https://discuss.circleci.com/t/container-step-shows-build-was-cancelled-and-is-a-missing-the-green-checkmark-although-all-tests-run-and-pass/37600

File: .circleci/config.yml (hunk 4 above)
Comment (pkfec) [reply .circleci/config.yml:21]: @rfultz - We had those **warning** prior to the postgres docker upgrade. With older docker version, the circleci would silently, ignore the warning and let the build run. 
Unless the ENV variable is specified, the builds will fail.

I posted the notes in my PR already on why the ENV variable need to set  and posting here again for your reference:https://discuss.circleci.com/t/postgresql-image-password-not-specified-issue/34555?utm_medium=SEM&utm_source=gnb&utm_campaign=SEM-gb-DSA-Eng-uscan&utm_content=&utm_term=dynamicSearch-&gclid=CjwKCAjw_NX7BRA1EiwA2dpg0hsoFeu7z02id7KLIrJiImZpUfDShDzw7DbAwJJdL75bstnqP4914RoCoWAQAvD_BwE

File: .circleci/config.yml (hunk 2 above)
Comment (pkfec) [reply .circleci/config.yml:54]: @rfultz Its the task name that we see on circleci builds. Do you prefer any other name? Please suggest.

File: .circleci/config.yml (hunk 1 above)
Comment (pkfec) [reply .circleci/config.yml:65]: @rfultz thanks for the suggestion. I will go ahead and remove the `npm install webpack` from the script.

File: .circleci/config.yml (hunk 2 above)
Comment (patphongs) [reply .circleci/config.yml:54]: @rfultz @pkfec What about something like this? "Install node and build web assets". Seems like a generic way of saying we're building everything from JS/scss/images

Comment (pkfec) [reply .circleci/config.yml:54]: @patphongs the proposed task name sounds good. I will go ahead and update it, unless there is any objection.

**Comment by pkfec** [comment]
@patphongs Like you suggested i update the Node task name to `Install node and build web assets`. Please review and let me know if you have any questions or concerns.
cc @rfultz @johnnyporkchops

**Review by rfultz — APPROVED** [review]

**Comment by jason-upchurch** [comment]
@pkfec @patphongs I'm merging this in preparation for innovation sprint as 2 of 3 reviewers have approved. Please let me know if I need to revert.

**Pull request merged**